mlx/backend/accelerate/primitives.cpp

```diff
@@ -255,6 +257,96 @@ void AsType::eval_cpu(const std::vector<array>& inputs, array& out) {
   eval(inputs, out);
 }
 
```

Comment:
Feels a bit wrong having this as a primitive but I wasn't sure if there's a better way to it.